src/config/config.ts

```diff
@@ -4,18 +4,45 @@ import { configSchema } from './config-schema';
 const safeParseConfig = (config: unknown) => {
   const result = configSchema.safeParse(config);
   if (!result.success) {
-    console.error('Invalid configuration:', result.error.format());
-    // improve error visibility in different environments
-    for (const issue of result.error.issues) {
+    console.error('Invalid configuration:');
+    result.error.issues.forEach(issue => {
       console.error(` - ${issue.path.join('.')} : ${issue.message}`);
-    }
+    });
     throw new Error('Invalid configuration');
   }
   return result.data;
 };
-const getConfig = () => {
-  return safeParseConfig(configIn);
+
+const appConfig = configIn.get('app') as Record<string, unknown> | undefined;
+const databaseConfig = configIn.get('database') as
+  | Record<string, unknown>
+  | undefined;
+const jwtConfig = configIn.get('JWT') as Record<string, unknown> | undefined;
+
+const nodeEnvFromConfig =
+  typeof appConfig?.nodeEnv === 'string' ? appConfig.nodeEnv : '';
+const nodeEnv =
+  nodeEnvFromConfig || process.env.ENV || process.env.NODE_ENV || 'development';
+
+const configToValidate = {
+  app: {
+    port: typeof appConfig?.port === 'number' ? appConfig.port : 3000,
+    nodeEnv: nodeEnv,
+    logLevel:
+      typeof appConfig?.logLevel === 'string' ? appConfig.logLevel : 'info',
+  },
+  database: {
+    url: typeof databaseConfig?.url === 'string' ? databaseConfig.url : '',
+  },
+  JWT: {
+    secret: typeof jwtConfig?.secret === 'string' ? jwtConfig.secret : '',
+    expiresIn:
```

Comment:
The database URL in configToValidate defaults to an empty string when not provided. Silently defaulting critical config (DB URL) to an empty string will likely cause confusing runtime errors later and could mask missing configuration.

**Sugerencia:**
```typescript
Fail fast when required configuration is missing. For example, if database URL is required, validate and throw an explicit error if it's absent instead of defaulting to ''. Example change:
url: typeof databaseConfig?.url === 'string' && databaseConfig.url ? databaseConfig.url : undefined
and then in safeParseConfig ensure database.url is required, or throw: throw new Error('Database URL is required').
```